[New Rule] Cisco Secure Endpoint Malicious File Detection

MakoWish opened the pull request:
## Issues
- Closes #2878 

## Summary
This detection rule queries for events indicating Cisco Secure Endpoint detected a malicious file.

## Contributor checklist

- [X] Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- [X] Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

MakoWish commented:
Test says the field `cisco.secure_endpoint.file.disposition` is unknown, but you can see the field exists in the sample data in: #2878

terrancedejesus commented:
👋🏽 Thanks for creating this rule!

Here is what needs to be done for unit tests to pass.
1. run `python -m detection_rules dev integrations build-manifests -o -i cisco_secure_endpoint` (adds `cisco_secure_endpoint` manifests to the `integration-manifests` file)
2. run `python -m detection_rules dev integrations build-schemas -o` (will add the `cisco_secure_endpoint` schema to the `integration-schemas` file)

Once this is done, the rule validation workflow should be able to find the integration specific fields in the integration schema, thus validating they exist.

MakoWish commented:
Conflicts are too complex? Never seen that before. Did I miss something? Or is that fixable on your end?

terrancedejesus commented:
> Conflicts are too complex? Never seen that before. Did I miss something? Or is that fixable on your end?

Rather odd. Did you by chance pull in all updates upstream into your fork before adjusting the manifest and schema files?

MakoWish commented:
I reverted back, updated my fork and branch from main, then I ran the commands to update the schemas and manifests, but it is still failing for the same field.

terrancedejesus commented on rules/integrations/cisco_secure_endpoint/cse_malicious_file_detection.toml:
```suggestion
event.category: malware and event.dataset: "cisco_secure_endpoint.event" and event.action: "threat detected" and cisco.secure_endpoint.file.disposition: Malicious
```

Whenever your data is based on a data stream from an integration, `event.dataset` has to be set.
Reference: https://github.com/elastic/integrations/blob/f9ef14cb1f6aa48a20eb60553ed1e02dd01561a7/packages/cisco_secure_endpoint/data_stream/event/fields/base-fields.yml#L20C5-L20C6

We have this error else where, but we should add it earlier.

MakoWish replied:
Do you know if there are plans to set `event.dataset` on all Integrations? I can see it is done here, but as it is now, `data_stream.dataset` seems to be the standard, and there are quite a lot where `event.dataset` is not utilized. 

I finally got this to pass locally. Just pushed the changes.

terrancedejesus reviewed (approved):
PR looks good, thanks for the addition!

MakoWish commented:
Not sure how to fix those conflicts now. I synchronized from main, checked out the manifest and schema files from `origin`, then rebuilt both locally, tested, and pushed. Tried twice, but still conflicting? Got any tips on how to resolve that?

MakoWish commented:
@terrancedejesus, any idea how to fix those conflicts?
